src/lazyload-image.directive.ts

```diff
@@ -1,44 +1,61 @@
 import 'rxjs/add/operator/let';
-import { Directive, ElementRef, Input, NgZone } from '@angular/core';
-import { getScrollListener } from './scroll-listener';
-import { lazyLoadImage } from './lazyload-image';
+import {Directive, ElementRef, Input, NgZone, SimpleChanges, OnChanges, OnDestroy} from '@angular/core';
+import {getScrollListener} from './scroll-listener';
+import {lazyLoadImage} from './lazyload-image';
+import {WindowService} from "./window.service";
+
+declare const global: any; // To make AoT compiler (ngc) happy
 
 @Directive({
-    selector: '[lazyLoad]'
+  selector: '[lazyLoad]'
 })
-export class LazyLoadImageDirective {
-    @Input('lazyLoad') lazyImage;   // The image to be lazy loaded
-    @Input() errorImage: string;    // The image to be displayed if lazyImage load fails
-    @Input() scrollTarget = window; // Chnage the node we should listen for scroll events on, default is window
-    @Input() scrollObservable;      // Pass your own scroll emitter
-    @Input() offset: number;        // The number of px a image should be loaded before it is in view port
-    elementRef: ElementRef;
-    ngZone: NgZone;
-    scrollSubscription;
-
-    constructor(el: ElementRef, ngZone: NgZone) {
-        this.elementRef = el;
-        this.ngZone = ngZone;
-    }
+export class LazyLoadImageDirective implements OnChanges, OnDestroy {
+  @Input('lazyLoad') lazyImage;   // The image to be lazy loaded
+  @Input() errorImage: string;    // The image to be displayed if lazyImage load fails
+  @Input() scrollTarget; // Change the node we should listen for scroll events on, default is window
+  @Input() scrollObservable;      // Pass your own scroll emitter
+  @Input() offset: number;        // The number of px a image should be loaded before it is in view port
+  elementRef: ElementRef;
+  ngZone: NgZone;
+  scrollSubscription;
+  window;
+
+  constructor(el: ElementRef, ngZone: NgZone, window: WindowService) {
```

Comment:
Okay so it is three reasons:
1. Decoupling from DOM 
2. Testability
3. The angular way

`1.` is the most important aspect but we can solve that with a ordinary function as well.
`2.`, sure, it may be easier to test (even though we can use the DI system for function as well) but right now we do not any test where we want to mock the `WindowService`.
`3.`, I agree, it is the Angular way to create a service class so I can agree to use a class in this case.